From bdc2b6676c0d20d19668ba4f08c39be79d9be5dc Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Thu, 2 Oct 2025 01:27:32 +0200
Subject: [PATCH] [Autofill] Use complete FormFieldData type for
 base::span<const FormFieldData>

Addresses build failure:
In file included from ../../components/autofill/core/browser/form_parsing/travel_field_parser.cc:5:
In file included from ../../components/autofill/core/browser/form_parsing/travel_field_parser.h:11:
In file included from ../../components/autofill/core/browser/form_parsing/autofill_scanner.h:11:
../../base/containers/span.h:1387:63: error: arithmetic on a pointer to an incomplete type 'element_type' (aka 'const autofill::FormFieldData')
 1387 |         typename iterator::AssumeValid(data(), data(), data() + size())));
      |                                                        ~~~~~~ ^
---
 .../autofill/core/browser/form_parsing/autofill_scanner.h      | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/components/autofill/core/browser/form_parsing/autofill_scanner.h b/components/autofill/core/browser/form_parsing/autofill_scanner.h
index 844526a5f77e1..eab073266e8bd 100644
--- a/components/autofill/core/browser/form_parsing/autofill_scanner.h
+++ b/components/autofill/core/browser/form_parsing/autofill_scanner.h
@@ -10,11 +10,10 @@
 #include "base/compiler_specific.h"
 #include "base/containers/span.h"
 #include "base/memory/raw_span.h"
+#include "components/autofill/core/common/form_field_data.h"
 
 namespace autofill {
 
-class FormFieldData;
-
 // A helper class for parsing a stream of |FormFieldData|'s with lookahead.
 class AutofillScanner {
  private:
